### PR TITLE
sensors/lsm6dsl: Fix SPI API usage

### LINKED ISSUE #7009: LSM6DSL: Isse with spi_config
Seems that since recent SPI API update, spi_config as lost dev element.

In LSM6DSL new SPI interface is not in line: in lsm6dsl_spi_init
``
	lsm6dsl_spi_conf.dev = data->comm_master;
``
This should be reviewed

### PR DESCRIPTION
**Pull request by tbursztyka**
The driver was not moved to the recent SPI API changes.

Fixes #7009

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

### CODE REVIEW

File: drivers/sensor/lsm6dsl/lsm6dsl_spi.c [hunk 1]
```diff
@@ -30,12 +30,17 @@ static struct spi_config lsm6dsl_spi_conf = {
 static int lsm6dsl_raw_read(struct spi_config *spi_cfg, u8_t reg_addr,
```
Comment (avisconti) [inline drivers/sensor/lsm6dsl/lsm6dsl_spi.c]: Should be:
static int lsm6dsl_raw_read(struct lsm6dsl_data *data, u8_t reg_addr,

File: drivers/sensor/lsm6dsl/lsm6dsl_spi.c [hunk 2]
```diff
@@ -60,8 +70,9 @@ static int lsm6dsl_raw_read(struct spi_config *spi_cfg, u8_t reg_addr,
 static int lsm6dsl_raw_write(struct spi_config *spi_cfg, u8_t reg_addr,
```
Comment (avisconti) [inline drivers/sensor/lsm6dsl/lsm6dsl_spi.c]: Should be:
static int lsm6dsl_raw_write(struct lsm6dsl_data *data, u8_t reg_addr,

File: drivers/sensor/lsm6dsl/lsm6dsl_spi.c [hunk 3]
```diff
@@ -30,12 +30,17 @@ static struct spi_config lsm6dsl_spi_conf = {
...
-	u8_t buffer_tx[2] = {reg_addr | LSM6DSL_SPI_READ, 0};
+	struct spi_config *spi_cfg = &lsm6dsl_spi_conf;
+	u8_t buffer_tx[2] = { reg_addr | LSM6DSL_SPI_READ, 0 };
 	const struct spi_buf tx_buf = {
 			.buf = buffer_tx,
 			.len = 2,
 	};
-	struct spi_buf rx_buf[2] = {
+	const struct spi_buf_set tx = {
+		.buffers = &tx_buf,
```
Comment (avisconti) [inline drivers/sensor/lsm6dsl/lsm6dsl_spi.c]: .buffers = tx_buf,

**Review by avisconti — COMMENTED** [review]
I fixed few typos to get it compiled.
Now I'm going to test it out.

**Comment by tbursztyka** [comment]
Applied the comments and fixed the commit title as well.

**Comment by avisconti** [comment]
@tbursztyka 
The a5b14e9 commit works fine on my setup! I didn't review it carefully, a part the changes done
to get it compiled. But I think it is good!

File: drivers/sensor/lsm6dsl/lsm6dsl_spi.c [hunk 4]
```diff
@@ -71,12 +82,17 @@ static int lsm6dsl_raw_write(struct spi_config *spi_cfg, u8_t reg_addr,
 			.len = len,
 		}
 	};
+	const struct spi_buf_set tx = {
+		.buffers = tx_buf,
+		.count = 1
```
Comment (avisconti) [inline drivers/sensor/lsm6dsl/lsm6dsl_spi.c]: shouldn't be this one equals to '2'?

Comment (tbursztyka) [reply drivers/sensor/lsm6dsl/lsm6dsl_spi.c]: yep, I should buy new glasses (can't change the brain so...)

Comment (avisconti) [reply drivers/sensor/lsm6dsl/lsm6dsl_spi.c]: @tbursztyka 
OK, thx!
The acc/gyro stuff was working fine.
Nevertheless I'm now re-testing (prior to submit) the new sensorhub part, and your changes
are not working anymore. If I put '2' there everything works.

Comment (tbursztyka) [reply drivers/sensor/lsm6dsl/lsm6dsl_spi.c]: pushed a fixed version

**Comment by avisconti** [comment]
9ff01c7 seems good also for the sensorhub part

**Comment by avisconti** [comment]
@erwango 
I reviewed and deeply tested this patch on ArgonKey: it is working fine.
Can you possibly take a look at it as well?

**Review by erwango — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**